marklogic-data-hub-central/ui/src/api/queries.tsx

```diff
@@ -60,6 +60,61 @@ export const exportQuery = (query, limit) => {
   mapForm.reset();
 }
 
+export const getExportQueryPreview = async (query) => {
+  let queryString = JSON.stringify(query)
+  const mapForm = document.createElement("form");
+  mapForm.id = "exportForm";
+  const mapInput = document.createElement("input");
+  mapInput.type = "hidden";
+  mapInput.name = "fileType";
+  mapInput.value = "csv";
+  const mapInput2 = document.createElement("input");
+  mapInput2.type = "hidden";
+  mapInput2.name = "limit";
+  mapInput2.value = '2';
+  const mapInput3 = document.createElement("input");
+  mapInput3.type = "hidden";
+  mapInput3.name = "queryDocument";
+  mapInput3.value = queryString;
+  mapForm.appendChild(mapInput);
+  mapForm.appendChild(mapInput2);
+  mapForm.appendChild(mapInput3);
+  document.body.appendChild(mapForm);
+
+  return new Promise(function (resolve, reject) {
+    let xhr = new XMLHttpRequest();
+    xhr.open("POST", "/api/entitySearch/export");
+    xhr.onload = function () {
+        if (this.status >= 200 && this.status < 300) {
```

Comment:
@timur-isangulov any reason for checking status less than 300 ?

Reply:
Checking for range of success status codes.